dwi_ml/data/processing/dwi/dwi.py

```diff
@@ -63,3 +62,52 @@ def standardize_data(data: np.ndarray, mask: np.ndarray = None,
     normalized_data[~mask] = np.nan
 
     return normalized_data
+
+
+def resample_raw_dwi_from_sh(dwi_image: nib.Nifti1Image,
+                             gradient_table: GradientTable,
+                             sphere: Sphere = None, sh_order: int = 8,
+                             smooth: float = 0.006):
+    """Resample a diffusion signal according to a set of directions using
+    spherical harmonics.
+
+    Parameters
+    ----------
+    dwi_image : nib.Nifti1Image object
+        Diffusion signal as weighted images (4D).
+    gradient_table : GradientTable
+        Dipy object that contains all bvals and bvecs.
+    sphere : dipy.core.sphere.Sphere, optional
+        Directions the diffusion signal will be resampled to. Directions are
+        assumed to be on the whole sphere, not the hemisphere like bvecs.
+        If omitted, 100 directions evenly distributed on the sphere will be
+        used (Dipy's "repulsion100").
+    sh_order : int, optional
+        SH order to fit, by default 8.
+    smooth : float, optional
+        Lambda-regularization coefficient in the SH fit, by default 0.006.
+
+    Returns
+    -------
+    resampled_dwi : np.ndarray (4D)
+        Resampled "raw" diffusion signal.
+    """
+    # Get SH fit
+    data_sh = compute_sh_coefficients(dwi_image, gradient_table, sh_order,
+                                      basis_type='tournier07', smooth=smooth)
```

Comment:
I would make basis_type an argument to the function, not hard-coded here... This is dangerous. 

And often, the SH basis will be descoteaux07